lnwallet/channel.go

```diff
@@ -3027,11 +3027,17 @@ func (lc *LightningChannel) fetchCommitmentView(remoteChain bool,
 	}
 	fee := lc.channelState.Capacity - totalOut
 
+	var witnessWeight int64
+	if lc.channelState.ChanType.IsTaproot() {
+		witnessWeight = input.TaprootKeyPathWitnessSize
```

Comment:
I'm confused. If the weight of a taproot key spend is smaller than the v0 multisig spend, wouldn't we have **over**estimated the fee instead?
Or is this when we check the proposed fee rate of the other party (which calculated it correctly), we reject it because we think it's too low?

Reply:
Hmm also swore I had this updated in one of the side branches, may have been dropped in a rebase conflict. Open question though is: why wasn't this caught by any of the itests or the unit tests?

Reply:
Dug a bit deeper, and it was missing in all the branches. We have unit tests to assert proper weight/fee calculation, but that's in another area when we construct the commitment transaction to then compare against the weight estimates.

Reply:
Re @guggero's questions above: iiuc, yes we would instead over estimate. This is called both when we go to propose a state, and also when we go to accept a state from the remote party. 

This seems independent of zero conf all together (since that was mentioned in the initial issue). One other relevant tidbit was: using mainnet fee estimation (neutrino) for a testnet node.

Reply:
I think the issue is, we use a much higher fee rate in itests, so this edge case was never caught. Added a new test case to check this edge case.